### DISCUSSION TITLE
Inconsistent file picker behaviour

### DISCUSSION THREAD

**Discussion by bbrodriges** [helix-editor/helix#8060] [General] [answered]
### Summary

File picker behaviour varies depending on presence of certain directories in project root (i.e. `.git` or `.helix` directories).

### Reproduction Steps

Consider some project with non-trivial structure and no `.git` folder in root:

```
$ tree
.
├── Dockerfile
├── Makefile
├── README.md
├── cmd
│   ├── sandboxtest
│   │   ├── main_test.go
│   │   ├── scenario_author_test.go
│   │   ├── scenario_gatekeeper_test.go
│   │   ├── scenario_testlib_test.go
│   │   ├── scenario_user_test.go
│   │   └── suite_test.go
│   └── server
│       ├── config.go
│       ├── instance.go
│       ├── main.go
│       └── router.go
├── container_config.json
├── fakemod
│   ├── fakemod.go
│   ├── go.mod
│   └── go.sum
├── go.mod
├── go.sum
└── internal
    ├── buffer
    │   ├── capped.go
    │   └── capped_test.go
    ├── gatekeeper
    │   ├── gatekeeper.go
    │   ├── rules.go
    │   └── rules_test.go
    ├── gomodule
    │   ├── action_build.go
    │   ├── action_build_test.go
    │   ├── actions.go
    │   ├── exec.go
    │   ├── gomodule.go
    │   └── module_opts.go
    ├── logger
    │   └── logger.go
    └── models
        ├── code_input.go
        └── result.go

```

Running `$ hx internal` will give us the following file picker state at startup:

<img width="1709"  src="https://github.com/helix-editor/helix/assets/871322/d59e36fe-882c-45fa-b834-0b9ad56a0f32">

If we close file picker with `Esc` and call it back again with `Space+f` we will see the same state.

But if we add `.git` (or `.helix`) to the project root than we'll see completely different behaviour:

https://github.com/helix-editor/helix/assets/871322/9c5bccf2-fd72-4c4f-9b3b-144a43dc8ad5

I've expected that file picker considers path passed to `hx` command as root directory. Instead root is detected automatically based on some internal logic.

### Platform

macOS

### Terminal Emulator

iTerm2 Build 3.4.20

### Helix Version

helix 23.05 (7f5940be)

**Comment by pascalkuthe** [comment, marked as answer]
This is intentional behavior and a feature. The file picker opens at the workspace root (which is marked by .helix or .git). If no workspace root is detected, cwd is used as a fallback

**Comment by gabydd** [comment]
You can use ``<space>F`` to search in the current working directory instead of the workspace root. You can also switch the ``f`` and ``F`` mappings if you prefer that behaviour.

**Reply by bbrodriges** [reply]
Thanks. I've become so used to `<space>f` that I've completely forgot about `<space>F` :)
